### ISSUE TITLE
Invalid SARIF: additionalProperty "fixes" exists in instance when not allowed

### ISSUE THREAD

**Issue by sim642** [semgrep/semgrep#5421]
**Describe the bug**
SARIF output produces invalid SARIF and thus fails to be uploaded to GitHub Security:
```
Error: Unable to upload "semgrep.sarif" as it is not valid SARIF:
- instance additionalProperty "fixes" exists in instance when not allowed
```

**To Reproduce**
I have the following GitHub actions workflow that fails with the above error at the upload step:
```yaml
jobs:
  semgrep:
    runs-on: ubuntu-latest

    container:
      image: returntocorp/semgrep

    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Run semgrep
        run: semgrep scan --sarif --output=semgrep.sarif

      - name: Upload SARIF file to GitHub Advanced Security Dashboard
        uses: github/codeql-action/upload-sarif@v2
        with:
          sarif_file: semgrep.sarif
        if: always()
```

**Expected behavior**
SARIF file should be valid and successfully uploadable to GitHub Security, as it previously was.

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
Docker image in GitHub Actions.

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1433/invalid-sarif-additionalproperty-fixes-exists-in-instance-when-not. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by sim642** [comment]
Semgrep generates a SARIF file, which begins with:
```json
{
    "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
    "fixes": [],
    "runs": [
```

According to SARIF 2.1.0, the `fixes` property is allowed for individual `result` objects, not at the top level:
https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317668.

Also that "$schema" link is a 404.

**Comment by kristof-mattei** [comment]
Reading about the scheme some more it seems that `fixes` should be a child of a `result` object. However in https://github.com/returntocorp/semgrep/commit/e8a3ad7db5c64827c160b7d83ca52aaf6a367211 it was introduced as a top-level attribute.

From the docs:

https://docs.oasis-open.org/sarif/sarif/v2.1.0/csprd01/sarif-v2.1.0-csprd01.html#_Toc10541106
> A result object MAY contain a property named fixes whose value is an array of zero or more unique (§3.7.3) fix objects (§3.55).


So we'll need to move the` fixes` to each individual result, not as an aggregate on top. 

WIP here: https://github.com/returntocorp/semgrep/pull/5425

**Comment by sim642** [comment]
Looks like this was fixed in #5425 and quietly released in v0.98.0 with no mention in the changelog. The CI jobs stopped failing.

**Comment by peteryates** [comment]
Worked fine for us too, thanks @sim642.
